Remove custom `Icon` component in favour of `@wordpress/icons` where possible

mikejolley opened the pull request:
This PR removes our `Icon` component in favour of the Icon component that is included with `@wordpress/icons`. On top of this, I've removed/migrated as many `@woocommerce/icons` as possible in favour of similar or equivalent `@wordpress/icons` in order to reduce our package size.

Fixes #3512

List of substitutions: 

| Woo Library | Usage | WP Library | 
|:---:|:-----:|:---:|
| atom | Product Elements/Atomic Blocks Section | woo |
| done | Cart button checkmark | check |
| folder | Block icon | file or archive |
| image | Block icon | image |
| star | Block icon | starEmpty |
| tag | Block icon | tag |
| grid | Block icon | grid |
| bill | Block icon | currencyDollar |
| box | Block icon | box |
| notes | Block icon | page |
| bookmark | Block icon | heading |
| arrowBack | Button | arrowLeft |
| noAlt | Chip component | closeSmall |
| card | Gateways | payment |
| bank | Gateways | institution |
| search | Search components | search |
| notice | Error message components | warning |
| discussion | Reviews widgets | postComments |
| chevronUp/chevronDown | Panels | chevronUp/chevronDown |
| external | External links in panels | external |
| server | Block icon for attributes | category |
| truck | Block icon | shipping |
| stonks | Block icon | trendingUp |
| address | Block icon | mapMarker |
| comment | Block icon | commentContent |
| asterisk | Terms and Conditions Block | customPostType |
| list | Block icon | list |
| review | Block icon | commentContent |
| reader | Single product block icon | mediaAndText |
| restore | Single Product Block | backup |
| contact | Contact Block Icon | atSymbol |
| tags | Products by attribute Block | category |
| more | Products by tag block | tag |
| exclamation | New products block icon | sparkles |
| widgets | Handpicked products block | category |

#### Other Checks

- [ ] We need to confirm the packages are available in supported WordPress versions before shipping
- [x] Need to confirm bundle size has not increased too much on frontend where icons are used

### Screenshots

| Before | After |
| ------------- | ------------- |
| <img width="248" alt="Screenshot 2022-02-01 at 12 36 17" src="https://user-images.githubusercontent.com/1628454/151965347-e6802d2f-6cb2-463a-8c7b-76b0594d8210.png"> |  <img width="250" alt="Screenshot 2022-02-01 at 13 04 10" src="https://user-images.githubusercontent.com/1628454/151971955-8907a318-b461-45ae-b586-e7db9b650284.png"> |
| <img width="251" alt="Screenshot 2022-02-01 at 13 06 50" src="https://user-images.githubusercontent.com/1628454/151965729-2dde8265-d323-4376-b392-de794bc50098.png"> |  <img width="252" alt="Screenshot 2022-02-01 at 13 05 45" src="https://user-images.githubusercontent.com/1628454/151965707-3b6141df-a83b-483d-9792-18d20347ea11.png"> |

### Testing

### Manual Testing

**Confirm there are no build errors** since this should pick up any mistakes with icon naming.

How to test the changes in this Pull Request:

1. Open block inserted and smoke test block icons
2. Try inner blocks within cart/checkout/single-product/all product blocks and confirm icons are working for inner blocks.
3. Check storybook works for icon library

No testing steps for end-users.

### Performance Impact

We need to confirm bundles sizes are not increased due to using the `@wordpress/icons` package. 

### Changelog

> Update block icons to use `@wordpress/icons` where possible.

mikejolley committed:
Remove atom icon

mikejolley committed:
Swap icon usage to WordPress package

mikejolley committed:
remove unused icons from library

mikejolley committed:
Use @wordpress/primitives

mikejolley committed:
Product cats block to listView

mikejolley committed:
On-sale to use percent instead of tag icon

mikejolley committed:
add to cart button use button icon

mikejolley committed:
on sale products block icon

mikejolley committed:
Handpicked products use stack icon

mikejolley committed:
Products by stock box icon

mikejolley committed:
Make sparkles woo purple

mikejolley committed:
Fix icon name

mikejolley committed:
Correct more tag name

mikejolley committed:
fix path to icons

mikejolley committed:
Update snaps

mikejolley commented:
Conflicts :/ 

I'll resolve them once this is reviewed.

ralucaStan commented on assets/js/icons/library/woo.tsx:
```diff
@@ -1,7 +1,7 @@
  * External dependencies
  */
-import { SVG } from 'wordpress-components';
+import { SVG } from '@wordpress/primitives';
```
There is a small leftover import in the project, it would be great if that could be replaced as well 
```
import { Path, SVG } from '@wordpress/components';
```

ralucaStan commented on assets/js/icons/README.md:
````diff
@@ -5,21 +5,14 @@ WooCommerce Blocks Icons Library.
 
 ```js
-import { Icon, bill, woo } from '@woocommerce/icons';
+import { woo } from '@woocommerce/icons';
````
It would be great if we had a small comment explaining why the two imports. 

Also, under `## Adding Icons` we could mention

0. Make sure the icon is not already included in [the Library](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) that comes with @wordpress/icons package.

ralucaStan reviewed (changes requested):
Huge effort here @mikejolley !
Code wise it looks good I made some small comments. 

Personally I liked the old All products better, but other than that I think most icon changes look better. I will post a comment with before and after image

ralucaStan commented:
The brach has some conflicts now :(

mikejolley commented:
@ralucaStan I am rebasing now. I knew it would get conflicts due to number of files changed.

mikejolley committed:
Update readme to explain updated usage

mikejolley committed:
Import SVG and path from primitives

ralucaStan commented:
@mikejolley I've used the images below to understand the changes, maybe add them to the description so that they could be used in the release post? 

| Before | After |
| ------------- | ------------- |
| <img width="248" alt="Screenshot 2022-02-01 at 12 36 17" src="https://user-images.githubusercontent.com/1628454/151965347-e6802d2f-6cb2-463a-8c7b-76b0594d8210.png"> |  <img width="250" alt="Screenshot 2022-02-01 at 13 04 10" src="https://user-images.githubusercontent.com/1628454/151971955-8907a318-b461-45ae-b586-e7db9b650284.png"> |
| <img width="251" alt="Screenshot 2022-02-01 at 13 06 50" src="https://user-images.githubusercontent.com/1628454/151965729-2dde8265-d323-4376-b392-de794bc50098.png"> |  <img width="252" alt="Screenshot 2022-02-01 at 13 05 45" src="https://user-images.githubusercontent.com/1628454/151965707-3b6141df-a83b-483d-9792-18d20347ea11.png"> |

mikejolley commented:
@ralucaStan thanks Ill update it. Your "after" image on the first row is not the correct one.

mikejolley committed:
Missing isFeaturePluginBuild

ralucaStan commented:
> @ralucaStan thanks Ill update it. Your "after" image on the first row is not the correct one.

fixed

mikejolley commented:
Thanks @ralucaStan I've updated the issue and fixed your suggestions. Should we merge as-is, or did you want to find an alternative for all products block? I went with a grid theme similar to what we had before.

ralucaStan commented on assets/js/icons/README.md:
```diff
@@ -4,26 +4,23 @@ WooCommerce Blocks Icons Library.
-import { Icon, bill, woo } from '@woocommerce/icons';
-
-<Icon srcElement={ bill } />
+Note we use hte `Icon` component from `@wordpress/icons`, and the Icon SVG from `@woocommerce/icons`.
```
```suggestion
Note we use the `Icon` component from `@wordpress/icons`. We use some SVG icons from `@woocommerce/icons` for WC Blocks specific icons, but we also use existing icons from `@wordpress/icons`.
```

ralucaStan commented on assets/js/icons/README.md:
````diff
@@ -4,26 +4,23 @@ WooCommerce Blocks Icons Library.
-<Icon srcElement={ bill } size={ 16 } />
+```js
+import { woo } from '@woocommerce/icons';
+import { Icon } from '@wordpress/icons';
````
```suggestion
import { Icon, postComments } from '@wordpress/icons';
```

ralucaStan commented on assets/js/icons/README.md:
```diff
@@ -4,26 +4,23 @@ WooCommerce Blocks Icons Library.
+import { Icon } from '@wordpress/icons';
 
-<Icon srcElement={ woo } width={ 20 } height={ Math.floor( 20 * 1.67 ) } />
+<Icon icon={ woo } />
```
```suggestion
<Icon icon={ woo } /> // icon  from '@woocommerce/icons'
<Icon icon={ postComments } /> // icon from '@wordpress/icons'
```

ralucaStan reviewed (approved):
I've added some suggestions for the readme update. 

I am ok to move fw with the All products icon you chose.

mikejolley committed:
Update assets/js/icons/README.md

Co-authored-by: Raluca Stan <raluca.stan@automattic.com>

mikejolley committed:
Update assets/js/icons/README.md

Co-authored-by: Raluca Stan <raluca.stan@automattic.com>

mikejolley committed:
Update assets/js/icons/README.md

Co-authored-by: Raluca Stan <raluca.stan@automattic.com>

mikejolley committed:
update lock
